certora/specs/CER-2-Operator/CER-68-Operator-authorization.spec

```diff
@@ -44,6 +44,30 @@ rule onlyOwnerCanCallSetOperator() {
     assert(getOperator(addressPrefix, operator) == operatorBitField);
 }
 
+rule onlyOwnerOrOperatorCanCallSetAccountOperator() {
+    env e;
+    address account;
+    address operator;
+    bool authorized;
+
+    // if msg.sender is the currentContract, it means we are within permit() and
+    // we need to use executionContext.getOnBehalfOfAccount() instead.
+    address actualCaller = e.msg.sender;
+    if (e.msg.sender == currentContract) {
+        actualCaller = getExecutionContextOnBehalfOfAccount(e);
+    }
+
+    // call the setAccountOperator method.
+    setAccountOperator(e, account, operator, authorized);
```

Comment:
I think that what we want to prove here is that only owner can **authorize** the operator (as per CER-68). if so, `setAccountOperator` should always be called with `true` and we should show that the function doesn't revert only if the actualCaller is an owner.

I also updated CER-52 which will cover `setAccountOperator` called with `false` (in that case for the call to be successful, the actualCaller will either be an owner or the operator deauthorizing itself)